## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 449312`
Commit: `e40c5a6f86fb82f1fba3148e8a24325f0096213b`

---
### `aarch64-linux`
<details>
  <summary>:x: 11 packages failed to build:</summary>
  <ul>
    <li>opensplatWithCuda</li>
    <li>osmscout-server</li>
    <li>python312Packages.deepsearch-glm</li>
    <li>python312Packages.deepsearch-glm.dist</li>
    <li>python312Packages.docling-parse</li>
    <li>python312Packages.docling-parse.dist</li>
    <li>python313Packages.deepsearch-glm</li>
    <li>python313Packages.deepsearch-glm.dist</li>
    <li>python313Packages.docling-parse</li>
    <li>python313Packages.docling-parse.dist</li>
    <li>tremotesf</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 10 packages built:</summary>
  <ul>
    <li>avizo</li>
    <li>bt-migrate</li>
    <li>cxxopts</li>
    <li>nwg-panel</li>
    <li>nwg-panel.dist</li>
    <li>opensplat</li>
    <li>pamixer</li>
    <li>protonmail-export</li>
    <li>sofa</li>
    <li>valhalla</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>python312Packages.deepsearch-glm</summary>
<pre>                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File &quot;/nix/store/xlza4p11vy30pjkhskaxyaxbwm2jpqsc-python3.12-pyproject-hooks-1.2.0/lib/python3.12/site-packages/pyproject_hooks/_in_process/_in_process.py&quot;, line 280, in build_wheel
    return _build_backend().build_wheel(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File &quot;/nix/store/srwxwvnir0lzmlqaf21gd4cy17455n8q-python3.12-poetry-core-2.2.0/lib/python3.12/site-packages/poetry/core/masonry/api.py&quot;, line 58, in build_wheel
    return WheelBuilder.make_in(
           ^^^^^^^^^^^^^^^^^^^^^
  File &quot;/nix/store/srwxwvnir0lzmlqaf21gd4cy17455n8q-python3.12-poetry-core-2.2.0/lib/python3.12/site-packages/poetry/core/masonry/builders/wheel.py&quot;, line 95, in make_in
    wb.build(target_dir=directory)
  File &quot;/nix/store/srwxwvnir0lzmlqaf21gd4cy17455n8q-python3.12-poetry-core-2.2.0/lib/python3.12/site-packages/poetry/core/masonry/builders/wheel.py&quot;, line 134, in build
    self._build(zip_file)
  File &quot;/nix/store/srwxwvnir0lzmlqaf21gd4cy17455n8q-python3.12-poetry-core-2.2.0/lib/python3.12/site-packages/poetry/core/masonry/builders/wheel.py&quot;, line 183, in _build
    self._run_build_script(self._package.build_script)
  File &quot;/nix/store/srwxwvnir0lzmlqaf21gd4cy17455n8q-python3.12-poetry-core-2.2.0/lib/python3.12/site-packages/poetry/core/masonry/builders/wheel.py&quot;, line 304, in _run_build_script
    subprocess.check_call([self.executable.as_posix(), build_script])
  File &quot;/nix/store/n9fq90ybwm7flzhswka4i8siss56g4hq-python3-3.12.11/lib/python3.12/subprocess.py&quot;, line 413, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command &#x27;[&#x27;/nix/store/n9fq90ybwm7flzhswka4i8siss56g4hq-python3-3.12.11/bin/python3.12&#x27;, &#x27;build.py&#x27;]&#x27; returned non-zero exit status 1.

ERROR Backend subprocess exited when trying to invoke build_wheel</pre>
</details>
<details>
<summary>python313Packages.deepsearch-glm</summary>
<pre>        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File &quot;/nix/store/mimdypy8nazgmhil170lrh3q3zfvy0js-python3.13-poetry-core-2.2.0/lib/python3.13/site-packages/poetry/core/masonry/builders/wheel.py&quot;, line 95, in make_in
    wb.build(target_dir=directory)
    ~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
  File &quot;/nix/store/mimdypy8nazgmhil170lrh3q3zfvy0js-python3.13-poetry-core-2.2.0/lib/python3.13/site-packages/poetry/core/masonry/builders/wheel.py&quot;, line 134, in build
    self._build(zip_file)
    ~~~~~~~~~~~^^^^^^^^^^
  File &quot;/nix/store/mimdypy8nazgmhil170lrh3q3zfvy0js-python3.13-poetry-core-2.2.0/lib/python3.13/site-packages/poetry/core/masonry/builders/wheel.py&quot;, line 183, in _build
    self._run_build_script(self._package.build_script)
    ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File &quot;/nix/store/mimdypy8nazgmhil170lrh3q3zfvy0js-python3.13-poetry-core-2.2.0/lib/python3.13/site-packages/poetry/core/masonry/builders/wheel.py&quot;, line 304, in _run_build_script
    subprocess.check_call([self.executable.as_posix(), build_script])
    ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File &quot;/nix/store/jj007xqxislbfphgwsiirsyxyif0dgss-python3-3.13.7/lib/python3.13/subprocess.py&quot;, line 419, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command &#x27;[&#x27;/nix/store/jj007xqxislbfphgwsiirsyxyif0dgss-python3-3.13.7/bin/python3.13&#x27;, &#x27;build.py&#x27;]&#x27; returned non-zero exit status 1.

ERROR Backend subprocess exited when trying to invoke build_wheel</pre>
</details>
<details>
<summary>tremotesf</summary>
<pre>[ 93%] Building CXX object src/CMakeFiles/tremotesf_objects.dir/tremotesf_dbus_generated/org.freedesktop.portal.Notification.cpp.o
[ 93%] Building CXX object src/CMakeFiles/tremotesf_objects.dir/tremotesf_dbus_generated/org.freedesktop.Notifications.cpp.o
[ 94%] Building CXX object src/CMakeFiles/tremotesf_objects.dir/tremotesf_dbus_generated/org.freedesktop.FileManager1.cpp.o
[ 95%] Building CXX object src/CMakeFiles/tremotesf_objects.dir/tremotesf_objects_autogen/WQVHVUYMN6/qrc_translations.cpp.o
[ 96%] Building CXX object src/CMakeFiles/tremotesf_objects.dir/tremotesf_objects_autogen/GFLEUOY6XJ/qrc_status.cpp.o
[ 97%] Building CXX object src/CMakeFiles/tremotesf_objects.dir/tremotesf_objects_autogen/EWIEGA46WW/qrc_resources.cpp.o
[ 97%] Built target tremotesf_objects
[ 97%] Built target tremotesf_autogen_timestamp_deps
[ 98%] Automatic MOC for target tremotesf
[ 98%] Built target tremotesf_autogen
[100%] Building CXX object src/CMakeFiles/tremotesf.dir/startup/main.cpp.o
[100%] Building CXX object src/CMakeFiles/tremotesf.dir/tremotesf_autogen/mocs_compilation.cpp.o
[100%] Linking CXX executable tremotesf
/nix/store/7s6vdx40nsh9vm0n1hd27sd8shs9jkp4-binutils-2.44/bin/ld: warning: libicuuc.so.76, needed by /nix/store/v57qliwajr3ba84slaf40shxv8yxqr2n-qtbase-6.9.2/lib/libQt6Core.so.6.9.2, may conflict with libicuuc.so.74
/nix/store/7s6vdx40nsh9vm0n1hd27sd8shs9jkp4-binutils-2.44/bin/ld: CMakeFiles/tremotesf_objects.dir/startup/commandlineparser.cpp.o: undefined reference to symbol &#x27;_ZN6icu_768ByteSinkD2Ev&#x27;
/nix/store/7s6vdx40nsh9vm0n1hd27sd8shs9jkp4-binutils-2.44/bin/ld: /nix/store/49c2z6b9vag5m55ahg4ahzfa5l0z4hy9-icu4c-76.1/lib/libicuuc.so.76: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [src/CMakeFiles/tremotesf.dir/build.make:309: src/tremotesf] Error 1
make[1]: *** [CMakeFiles/Makefile2:349: src/CMakeFiles/tremotesf.dir/all] Error 2
make: *** [Makefile:136: all] Error 2</pre>
</details>
</details>

--- a/pkgs/by-name/cx/cxxopts/package.nix
+++ b/pkgs/by-name/cx/cxxopts/package.nix
@@ -2,6 +2,7 @@
   lib,
   stdenv,
   fetchFromGitHub,
+  fetchpatch2,
   cmake,
   icu74,
   pkg-config,
@@ -19,7 +20,7 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-baM6EX9D0yfrKxuPXyUUV9RqdrVLyygeG6x57xN8lc4=";
   };
 
-  buildInputs = lib.optionals enableUnicodeHelp [ icu74.dev ];
+  propagatedBuildInputs = lib.optionals enableUnicodeHelp [ icu74.dev ];
   cmakeFlags = [
     "-DCXXOPTS_BUILD_EXAMPLES=OFF"
   ]
@@ -36,6 +37,14 @@ stdenv.mkDerivation (finalAttrs: {
     substituteInPlace packaging/pkgconfig.pc.in \
       --replace '$'{prefix}/@CMAKE_INSTALL_INCLUDEDIR@ @CMAKE_INSTALL_FULL_INCLUDEDIR@
   '';
+
+  patches = [
+    (fetchpatch2 {
+      url = "https://github.com/jarro2783/cxxopts/commit/e98c73d665915b292a0592bf34fcbe8522035bc1.patch?full_index=1";
+      name = "fix-icu-uc-typo-in-pkgconfig.patch";
+      hash = "sha256-bqd3H66Op1/EkN2HLd84Obky4Y2ndPPY8MGZ5fqtdk4=";
+    })
+  ];
 
   meta = with lib; {
     homepage = "https://github.com/jarro2783/cxxopts";

--- a/pkgs/by-name/cx/cxxopts/package.nix
+++ b/pkgs/by-name/cx/cxxopts/package.nix
@@ -35,7 +35,7 @@ stdenv.mkDerivation (finalAttrs: {
   # https://github.com/jarro2783/cxxopts/issues/332
   postPatch = ''
     substituteInPlace packaging/pkgconfig.pc.in \
-      --replace '$'{prefix}/@CMAKE_INSTALL_INCLUDEDIR@ @CMAKE_INSTALL_FULL_INCLUDEDIR@
+      --replace-fail '$'{prefix}/@CMAKE_INSTALL_INCLUDEDIR@ @CMAKE_INSTALL_FULL_INCLUDEDIR@
   '';
 
   patches = [

--- a/pkgs/by-name/cx/cxxopts/package.nix
+++ b/pkgs/by-name/cx/cxxopts/package.nix
@@ -6,6 +6,8 @@
   cmake,
   icu74,
   pkg-config,
+  testers,
+  validatePkgConfig,
   enableUnicodeHelp ? true,
 }:
 
@@ -25,7 +27,13 @@ stdenv.mkDerivation (finalAttrs: {
     "-DCXXOPTS_BUILD_EXAMPLES=OFF"
   ]
   ++ lib.optional enableUnicodeHelp "-DCXXOPTS_USE_UNICODE_HELP=TRUE";
-  nativeBuildInputs = [ cmake ] ++ lib.optionals enableUnicodeHelp [ pkg-config ];
+  nativeBuildInputs = [
+    cmake
+  ]
+  ++ lib.optionals enableUnicodeHelp [
+    pkg-config
+    validatePkgConfig
+  ];
 
   doCheck = true;
 
@@ -46,11 +54,19 @@ stdenv.mkDerivation (finalAttrs: {
     })
   ];
 
+  passthru = {
+    tests.pkg-config = testers.hasPkgConfigModules {
+      package = finalAttrs.finalPackage;
+      versionCheck = true;
+    };
+  };
+
   meta = with lib; {
     homepage = "https://github.com/jarro2783/cxxopts";
     description = "Lightweight C++ GNU-style option parser library";
     license = licenses.mit;
     maintainers = [ maintainers.spease ];
+    pkgConfigModules = [ "cxxopts" ];
     platforms = platforms.all;
   };
 })

